kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/DeltaTableWritesSuite.scala

```diff
@@ -166,6 +170,108 @@ class DeltaTableWritesSuite extends DeltaTableWriteSuiteBase with ParquetSuiteBa
+      )
+      val ver1Snapshot = table.getLatestSnapshot(engine).asInstanceOf[SnapshotImpl]
+      assertMetadataProp(ver1Snapshot, TableConfig.CHECKPOINT_INTERVAL, 2)
+    }
+  }
+
+  test("create table and configure properties with retries") {
+    withTempDirAndEngine { (tablePath, engine) =>
+      // Create table
+      val table = Table.forPath(engine, tablePath)
+      createTxn(engine, tablePath, isNewTable = true, testSchema, Seq.empty)
+        .commit(engine, emptyIterable())
+      // Create txn1 with config changes
+      val txn1 = createTxn(
+        engine,
+        tablePath,
+        tableProperties = Map(TableConfig.CHECKPOINT_INTERVAL.getKey -> "2"))
+      // Create and commit txn2
+      appendData(
+        engine,
+        tablePath,
+        data = Seq(Map.empty[String, Literal] -> dataBatches1)
+      )
+
+      val ver1Snapshot = table.getLatestSnapshot(engine).asInstanceOf[SnapshotImpl]
+      assertMetadataProp(ver1Snapshot, TableConfig.CHECKPOINT_INTERVAL, 10)
+
+      // Try to commit txn1
+      txn1.commit(engine, emptyIterable())
+
+      val ver2Snapshot = table.getLatestSnapshot(engine).asInstanceOf[SnapshotImpl]
+      assertMetadataProp(ver2Snapshot, TableConfig.CHECKPOINT_INTERVAL, 2)
+    }
+  }
+
+  test("create table and configure the same properties") {
+    withTempDirAndEngine { (tablePath, engine) =>
+      val table = Table.forPath(engine, tablePath)
+      setTablePropAndVerify(
+        engine, tablePath, isNewTable = true, TableConfig.CHECKPOINT_INTERVAL, "2", 2)
+      assert(getMetadataActionFromCommit(engine, table, 0).isPresent)
+
+      appendData(
+        engine,
+        tablePath,
+        data = Seq(Map.empty[String, Literal] -> dataBatches1),
+        tableProperties =
+          Map(TableConfig.CHECKPOINT_INTERVAL.getKey.toLowerCase(Locale.ROOT) -> "2"))
+      val ver1Snapshot = table.getLatestSnapshot(engine).asInstanceOf[SnapshotImpl]
+      assertMetadataProp(ver1Snapshot, TableConfig.CHECKPOINT_INTERVAL, 2)
+      assert(!getMetadataActionFromCommit(engine, table, 1).isPresent)
+    }
+  }
+
+  test("create table - invalid properties - expect failure") {
+    withTempDirAndEngine { (tablePath, engine) =>
+      val ex1 = intercept[UnknownConfigurationException] {
+        createTxn(
+          engine, tablePath, isNewTable = true, testSchema, Seq.empty, Map("invalid key" -> "10"))
+      }
+      assert(ex1.getMessage.contains("Unknown configuration was specified: invalid key"))
+
+      val ex2 = intercept[InvalidConfigurationValueException] {
+        createTxn(
+          engine,
+          tablePath,
+          isNewTable = true,
+          testSchema, Seq.empty, Map(TableConfig.CHECKPOINT_INTERVAL.getKey -> "-1"))
+      }
+      assert(
+        ex2.getMessage.contains(
+          String.format(
+            "Invalid value for table property '%s': '%s'. %s",
+            TableConfig.CHECKPOINT_INTERVAL.getKey, "-1", "needs to be a positive integer.")))
+    }
+  }
```

Comment:
can you add a test verifying that the case of the property is same as the one in `TableConfig` and not the one passed by the user.